### ISSUE TITLE
Provide the tool for collecting crash artefacts the right way

### ISSUE THREAD

**Issue by igormunkin** [tarantool/tarantool#5569]
To ease the investigation process it would be nice, if all customers share their coredumps the single way: an archive with coredump, the corresponding binary and all related libraries. I have an ad-hoc script for such purposes, but I believe it's better to make a user-friendly tool (e.g. checking whether `gdb` is installed) from the one below to be shipped via our so-called "devel" binary package.

```bash
#!/bin/sh

PROGRAMMFILE=$1
COREFILE=$2

ARCHIVENAME=tarantool-core-$(date +%Y%m%d%H%M)-${HOSTNAME%%.*}.tar.gz
HOSTNAME=$(hostname -f)
TMPDIRECTORY=$(mktemp -d)
SHAREDLIST=${TMPDIRECTORY}/shared_libraries
SEPARATOR1="Shared Object Library"
SEPARATOR2="Shared library is missing debugging information"

/usr/bin/gdb -batch ${PROGRAMMFILE} -c ${COREFILE} -ex "info shared" -ex "quit" | \
        sed -n "/${SEPARATOR1}/,/${SEPARATOR2}/p;/${SEPARATOR2}/q" | \
        awk '{ print $NF }' | grep "^/" > ${SHAREDLIST}

echo $PROGRAMMFILE >> ${SHAREDLIST}
echo $COREFILE >> ${SHAREDLIST}

tar -czhf ${ARCHIVENAME} -T ${SHAREDLIST}

[[ -f "${SHAREDLIST}" ]] && rm -f "${SHAREDLIST}"
[[ -d "${TMPDIRECTORY}" ]] && rmdir "${TMPDIRECTORY}"

[[ $? == 0 ]] && echo "Archive: ${ARCHIVENAME}"
```

Anyway, now I can share this script via this issue until it's resolved.

**Comment by igormunkin** [comment]
I've polished and sent [the draft](https://lists.tarantool.org/pipermail/tarantool-patches/2021-February/022311.html) for the desired tooling:
* `tarabrt.sh` -- this is the script collecting a tarball with the coredump the right way (similar to the one above): with the binary, all loaded shared libs, Tarantool version (this is a separate exercise to get it from the binary built with -O2). Besides, I made the tarball layout unified so it can be easily processed with the second script:
  - the coredump -- `coredump` file on the root level
  - the binary -- `tarantool` executable on the root level
  - the version -- `version` plain text file on the root level with `tarantool --version` output
  - the tarball contents -- `checklist` plain text file on the root level with the list of the collected entities
  - all shared libraries used by the crashed instance -- their layout respects the one on the host machine, so them can be easily loaded with the following `gdb` command: `set sysroot $(realpath .)`
  
  The script can be easily used either manually or via `kernel.core_pattern` variable.
* `gdb.sh` -- the auxiliary script originally written by @Totktonada, that needed to be adjusted to the customer artefacts every time. Since we will have a unified layout I enhanced it a bit to automatically load the coredump via `gdb` the right way.
